plugin/trino-redshift/src/test/java/io/trino/plugin/redshift/TestRedshiftConnectorTest.java

```diff
@@ -684,6 +705,42 @@ protected SqlExecutor onRemoteDatabase()
         return TestingRedshiftServer::executeInRedshift;
     }
 
+    private SqlExecutor onRemoteDatabaseWithSchema(String schema)
+    {
+        return sql -> TestingRedshiftServer.executeInRedshift("set search_path to %s; ".formatted(schema) + sql);
+    }
+
+    @Override
+    protected void startTracingDatabaseEvent(RemoteLogTracingEvent event)
+    {
+        remoteDatabaseEventMonitor.startTracingDatabaseEvent(event);
```

Comment:
since you extracted TestingRedshiftServer, why not move all the tracing related code there?